prov/verbs/src/verbs_info.c

```diff
@@ -361,7 +362,12 @@ int fi_ibv_check_rx_attr(const struct fi_rx_attr *attr,
 		return -FI_ENODATA;
 	}
 
-	if (attr->total_buffered_recv > info->rx_attr->total_buffered_recv) {
+	rm_enabled =(info->domain_attr &&
+		     info->domain_attr->resource_mgmt == FI_RM_ENABLED);
+
+	if (!rm_enabled &&
+	    (attr->total_buffered_recv > info->rx_attr->total_buffered_recv))
+	{
```

Comment:
The value is an in/out parameter.  It is independent from FI_RM_ENABLED.

When FI_RM_ENABLED is set to 1, sent data transfers will not be lost (baring any fatal error).  This is true whether or not there is extra buffering on the receive side (i.e. total_buffered_recv > 0).

If FI_RM_ENABLED is set to 0, then any message that overruns the receive side buffering will be lost.  This is true for any value of total_buffered_recv.  The man pages is simply calling out that if total_buffered_recv = 0, then the very first message that doesn't match with a user buffer will be lost.

Total_buffered_recv is an in/out parameter.  The comment that the amount of buffering may be greater just means that the provider is not restricted by this limit.  For example, if the provider implements a rendezvous protocol as part of its unexpected message handing, then it could _appear_ to the application that it buffers more data than what's specified for total_buffered_recv.  However, this is provider specific and the app cannot rely on this.

Hopefully this makes sense.  If there's any wording change that you'd like to the man page to clarify things, please let me know.